slides/week3/git.org

```diff
@@ -0,0 +1,200 @@
+#+TITLE: Git Training - Week 3
+#+AUTHOR: Jay Kamat, Josh Ting
+#+EMAIL: jaygkamat@gmail.com, josh.ting@gatech.edu
+
+* What You Know
+- ~git status~
+- ~git add~
+- ~git commit~
+- ~git remote~
+- ~git pull~
+- ~git push~
+- /(any questions?)/
+
+* Some Problems
+- Multiple developers work on the same code, at the same time
+- We want to maintain multiple versions of the same software in parallel
+  - How do I apply a security fix to an old version?
+- You want a safe place to test out changes, without affecting anyone else
+- Branching is ~git~'s solution
+
+* Branches
+- Branches are local copies of state
+- If you do work on a branch, it won't affect any other branches
+- Later on, we can 'merge' branches together (so work done on either can be combined).
+- Branches are what people have the most trouble with
+  - _*Ask questions /please/!*_ This is a difficult topic.
+
+** Git As a Graph
+
+- git history is a *graph*
+- Each commit is a node on the graph
+- Ways to see it as a graph, rather than a line of commits
+  - ~git~ porcelain
+    #+BEGIN_SRC sh
+      git log --oneline --graph --all
+    #+END_SRC
+
+    #+BEGIN_SRC sh :results output :exports results
+      git log --oneline --graph | head -n10
+    #+END_SRC
+  - ~gitk~
+  - ~git~ integrations for your favorite editor!
+
+** Creating Branches
+- Let's start by creating multiple branches
+
+  #+BEGIN_SRC
+    844a162 * master stable beta Initial commit
```

Comment:
Hmm, it looks like a bunch of bugs relating to `--decorate` were fixed in git 2.4.3. Maybe you have an older version of git installed? Either way, it dosen't matter too much to this excersize if it dosen't work for others (since people can use graphical viewers like gitk, git gui, tig, etc). Hopefully people will be confident enough at this point to find something that works for them.